components/user_settings/notifications/user_settings_notifications.tsx

```diff
@@ -104,53 +151,21 @@ function getNotificationsStateFromProps(props) {
         autoResponderMessage,
         notifyCommentsLevel: comments,
         isSaving: false,
+        markUnread,
+        customcheckChecked: false,
     };
 }
 
-export default class NotificationsTab extends React.PureComponent {
-    static propTypes = {
-        user: PropTypes.object,
-        updateSection: PropTypes.func,
-        activeSection: PropTypes.string,
-        closeModal: PropTypes.func.isRequired,
-        collapseModal: PropTypes.func.isRequired,
-        sendPushNotifications: PropTypes.bool,
-        enableAutoResponder: PropTypes.bool,
-        actions: PropTypes.shape({
-            updateMe: PropTypes.func.isRequired,
-        }).isRequired,
-    }
-
-    static defaultProps = {
-        user: null,
-        activeSection: '',
-        activeTab: '',
-    }
-
-    constructor(props) {
-        super(props);
+export default class NotificationsTab extends React.PureComponent<Props, State> {
+    private custommentions: React.RefObject<HTMLInputElement>;
+    constructor({activeSection = '', ...restProps}: Props) {
```

Comment:
Instead of doing this to specify a default value for `activeSection`, you could instead leave `defaultProps` as it was before, although you'll need to make its type `Partial<Props>`.